kubeflow/trainer/rhai/transformers.py

```
@@ -463,8 +471,197 @@ def relative_update(self, inc=1):
+            staging_path, checkpoint_name, total_size, is_global_rank_0 = task
+            incomplete_marker_path = f"{checkpoint_name}/{CHECKPOINT_INCOMPLETE_MARKER}"
+
+            # Create .incomplete sentinel so resume can skip in-flight uploads.
+            if is_global_rank_0:
+                _log(f"Creating .incomplete marker in S3: {checkpoint_name}")
+            try:
+                self.remote_fs.pipe(
+                    incomplete_marker_path,
+                    f"Upload started for {checkpoint_name}".encode(),
+                )
+            except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)